### PR TITLE
Define k8s_best_practices_certsuite role

### PR DESCRIPTION
**Pull request by ramperher**
- Move cnf_cert role logic to new k8s_best_practices_certsuite, to follow the new naming convention of the cnf-certification-test project
- Added a fix in both roles, cnf_cert and k8s_best_practices_certsuite, for retrieving the debug-partner image, based on version.json file, when testing changes from stable releases. This is because, for example, debug-partner image for v5.2.2 is v5.2.1, so it's not true anymore that certsuite version could be used to retrieve the correct debug-partner image

Test-Hints: no-check

Testing:

- [x] certsuite v5.2.2 using legacy cnf_cert role in disconnected mode (to confirm stable versions are still working for legacy role) - https://www.distributed-ci.io/jobs/f34ecd32-f607-4cf2-8f5e-6c77b271e559/jobStates
- [x] certsuite HEAD version using new role in disconnected mode (this is used for testing PRs in cnf-certification-test repo) - https://www.distributed-ci.io/jobs/a3575bcb-4cbd-4f28-b208-1bff41a740bc/jobStates?sort=date
- [x] certsuite v5.2.2 using new role in disconnected mode - https://www.distributed-ci.io/jobs/7ed62949-9ea1-490f-8901-485f43c38bd8/jobStates?sort=date
- [x] certsuite v5.2.2 using new role in connected mode - https://www.distributed-ci.io/jobs/33254a5b-d9ad-40f2-b15f-0d014ffcda07/jobStates?sort=date
- [x] certsuite change (https://github.com/test-network-function/cnf-certification-test/pull/2264) using new role in disconnected mode, to confirm dci-pipeline update is working fine - https://www.distributed-ci.io/jobs/5670930f-1475-4718-8312-c43ce78d15c1/jobStates?sort=date
- [x] example-cnf using certsuite v5.2.2 and new role in disconnected mode - https://www.distributed-ci.io/jobs/71252887-50d2-4ea0-a5ae-d42e407b0184/jobStates/?sort=date

### CODE REVIEW

**Review by tkrishtop — DISMISSED** [review]

**Comment by ramperher** [comment]
@tonyskapunk , as commented before, here are the errors from ansible-lint right now:

```
-ERROR: roles/k8s_best_practices_certsuite/tasks/pre-run.yml:: redhat-ci[no-role-prefix]: Variable names from within roles should use a prefix related to the role. ({'k8s_best_practices_certsuite', 'kbpc', 'global'} are possible) (vars: authfile)
-ERROR: roles/k8s_best_practices_certsuite/tasks/pre-run.yml:: redhat-ci[no-role-prefix]: Variable names from within roles should use a prefix related to the role. ({'k8s_best_practices_certsuite', 'kbpc', 'global'} are possible) (vars: dci_local_registry)
-ERROR: roles/k8s_best_practices_certsuite/tasks/pre-run.yml:: redhat-ci[no-role-prefix]: Variable names from within roles should use a prefix related to the role. ({'k8s_best_practices_certsuite', 'kbpc', 'global'} are possible) (vars: images)
-ERROR: roles/k8s_best_practices_certsuite/tasks/pre-run.yml:: redhat-ci[no-role-prefix]: Variable names from within roles should use a prefix related to the role. ({'k8s_best_practices_certsuite', 'kbpc', 'global'} are possible) (vars: ic_dev_gits)
-ERROR: roles/k8s_best_practices_certsuite/tasks/pre-run.yml:: redhat-ci[no-role-prefix]: Variable names from within roles should use a prefix related to the role. ({'k8s_best_practices_certsuite', 'kbpc', 'global'} are possible) (vars: ic_gits)
```

The affected variables are variables used to call other roles:

- mirror_images for images, authfile and dci_local_registry vars
- include_components for ic_gits and ic_dev_gits

These errors should be omitted somehow because we're using the correct names for the variables since they're related to these roles. Could you take a look at it?

In the meantime, I'll start testing the role

**Comment by tonyskapunk** [comment]
> @tonyskapunk , as commented before, here are the errors from ansible-lint right now:
> 
> ```
> -ERROR: roles/k8s_best_practices_certsuite/tasks/pre-run.yml:: redhat-ci[no-role-prefix]: Variable names from within roles should use a prefix related to the role. ({'k8s_best_practices_certsuite', 'kbpc', 'global'} are possible) (vars: authfile)
> -ERROR: roles/k8s_best_practices_certsuite/tasks/pre-run.yml:: redhat-ci[no-role-prefix]: Variable names from within roles should use a prefix related to the role. ({'k8s_best_practices_certsuite', 'kbpc', 'global'} are possible) (vars: dci_local_registry)
> -ERROR: roles/k8s_best_practices_certsuite/tasks/pre-run.yml:: redhat-ci[no-role-prefix]: Variable names from within roles should use a prefix related to the role. ({'k8s_best_practices_certsuite', 'kbpc', 'global'} are possible) (vars: images)
> -ERROR: roles/k8s_best_practices_certsuite/tasks/pre-run.yml:: redhat-ci[no-role-prefix]: Variable names from within roles should use a prefix related to the role. ({'k8s_best_practices_certsuite', 'kbpc', 'global'} are possible) (vars: ic_dev_gits)
> -ERROR: roles/k8s_best_practices_certsuite/tasks/pre-run.yml:: redhat-ci[no-role-prefix]: Variable names from within roles should use a prefix related to the role. ({'k8s_best_practices_certsuite', 'kbpc', 'global'} are possible) (vars: ic_gits)
> ```
> 
> The affected variables are variables used to call other roles:
> 
>     * mirror_images for images, authfile and dci_local_registry vars
>     * include_components for ic_gits and ic_dev_gits

Agreed that their use is correct, when calling other roles.
At the same time this highlights what was discussed long ago when we migrated to this collection. We discussed that is likely that some roles are suitable for DCI and not for this collection, for instance those related to certifications. Mostly, because those use some of the roles in this collection and is unlikely to be used on their own. Perhaps this is good moment to discuss this topic, since is a new role. Anyway, this is food for thought.

Back to the issue you're facing you can add an exclusion on those vars to avoid the validation of that rule with `# noqa: redhat-ci[no-role-prefix]`

cc: @thekad

**Comment by thekad** [comment]
> > @tonyskapunk , as commented before, here are the errors from ansible-lint right now:
> > ```
> > -ERROR: roles/k8s_best_practices_certsuite/tasks/pre-run.yml:: redhat-ci[no-role-prefix]: Variable names from within roles should use a prefix related to the role. ({'k8s_best_practices_certsuite', 'kbpc', 'global'} are possible) (vars: authfile)
> > -ERROR: roles/k8s_best_practices_certsuite/tasks/pre-run.yml:: redhat-ci[no-role-prefix]: Variable names from within roles should use a prefix related to the role. ({'k8s_best_practices_certsuite', 'kbpc', 'global'} are possible) (vars: dci_local_registry)
> > -ERROR: roles/k8s_best_practices_certsuite/tasks/pre-run.yml:: redhat-ci[no-role-prefix]: Variable names from within roles should use a prefix related to the role. ({'k8s_best_practices_certsuite', 'kbpc', 'global'} are possible) (vars: images)
> > -ERROR: roles/k8s_best_practices_certsuite/tasks/pre-run.yml:: redhat-ci[no-role-prefix]: Variable names from within roles should use a prefix related to the role. ({'k8s_best_practices_certsuite', 'kbpc', 'global'} are possible) (vars: ic_dev_gits)
> > -ERROR: roles/k8s_best_practices_certsuite/tasks/pre-run.yml:: redhat-ci[no-role-prefix]: Variable names from within roles should use a prefix related to the role. ({'k8s_best_practices_certsuite', 'kbpc', 'global'} are possible) (vars: ic_gits)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The affected variables are variables used to call other roles:
> > ```
> > * mirror_images for images, authfile and dci_local_registry vars
> > * include_components for ic_gits and ic_dev_gits
> > ```
> 
> Agreed that their use is correct, when calling other roles. At the same time this highlights what was discussed long ago when we migrated to this collection. We discussed that is likely that some roles are suitable for DCI and not for this collection, for instance those related to certifications. Mostly, because those use some of the roles in this collection and is unlikely to be used on their own. Perhaps this is good moment to discuss this topic, since is a new role. Anyway, this is food for thought.
> 
> Back to the issue you're facing you can add an exclusion on those vars to avoid the validation of that rule with `# noqa: redhat-ci[no-role-prefix]`
> 
> cc: @thekad

@ramperher I see the problem, when you include_role the variable names would need to match the name of the included role, not the name of the role _including_ it. Still, that would put you in a spot where you'd have to go in and fix the other role first before passing your PR, which is cumbersome. I am with Tony just mark that task with noqa and we can fix the mirror_images role later. I will look into fixing the lint rule.

**Comment by ramperher** [comment]
Testing is finished and this is ready for review.

File: ansible-collection-redhatci-ocp.spec
Comment (thekad) [inline ansible-collection-redhatci-ocp.spec:55]: add deprecation here

File: roles/k8s_best_practices_certsuite/tasks/logging.yml
Comment (thekad) [inline roles/k8s_best_practices_certsuite/tasks/logging.yml:19]: kbpc_log_path | length > 0

File: roles/k8s_best_practices_certsuite/tasks/pre-run.yml
Comment (thekad) [inline roles/k8s_best_practices_certsuite/tasks/pre-run.yml:14]: if this role is new then "no longer compatible" makes no sense and also it doesn't *what* isn't compatible with, change to something like "This role is not compatible with X version < v.5.1.3"

File: roles/k8s_best_practices_certsuite/tasks/pre-run.yml
Comment (thekad) [inline roles/k8s_best_practices_certsuite/tasks/pre-run.yml:20]: the default is "all" does that include the preflight label? if so you gotta account for both cases

File: roles/k8s_best_practices_certsuite/tasks/pre-run.yml
Comment (thekad) [inline roles/k8s_best_practices_certsuite/tasks/pre-run.yml:167]: nitpick, please ignore: you've been using the ternary filter throughout the code

**Review by thekad — CHANGES_REQUESTED** [review]

File: ansible-collection-redhatci-ocp.spec
Comment (ramperher) [reply ansible-collection-redhatci-ocp.spec:55]: ack

File: README.md
Comment (ramperher) [reply README.md:47]: ack

File: roles/k8s_best_practices_certsuite/tasks/logging.yml
Comment (ramperher) [reply roles/k8s_best_practices_certsuite/tasks/logging.yml:19]: ack

File: roles/k8s_best_practices_certsuite/tasks/pre-run.yml
Comment (ramperher) [reply roles/k8s_best_practices_certsuite/tasks/pre-run.yml:14]: good catch, this is because I just copied the same task from cnf_cert role, I'll change that

File: roles/k8s_best_practices_certsuite/tasks/pre-run.yml
Comment (ramperher) [reply roles/k8s_best_practices_certsuite/tasks/pre-run.yml:20]: I'm not 100% sure that preflight is launched if only providing `all`. For the moment, I'll accept your suggestion and retest; if preflight is not launched if just using `all`, then I'll revert

File: roles/k8s_best_practices_certsuite/tasks/pre-run.yml
Comment (ramperher) [reply roles/k8s_best_practices_certsuite/tasks/pre-run.yml:167]: good catch, again this task is inherited from the previous cnf_cert role, fixed

File: roles/k8s_best_practices_certsuite/tasks/pre-run.yml
Comment (ramperher) [reply roles/k8s_best_practices_certsuite/tasks/pre-run.yml:22]: ack

File: roles/k8s_best_practices_certsuite/tasks/pre-run.yml
Comment (ramperher) [reply roles/k8s_best_practices_certsuite/tasks/pre-run.yml]: @thekad preflight is not launched if using `all` label, checked here: https://www.distributed-ci.io/jobs/8fdfa989-eccd-4e44-a817-1e7c481b6f8b/tests/6358204a-e7d2-4dd0-91f9-972ac271eefa, so I'll return to the prev code, this is not needed

**Comment by ramperher** [comment]
Change is ready for review again.

File: roles/k8s_best_practices_certsuite/tasks/tests.yml
Comment (tkrishtop) [inline roles/k8s_best_practices_certsuite/tasks/tests.yml:18]: should it be 0755 as well?

**Review by tkrishtop — DISMISSED** [review]
LGTM, one minor question about the template permissions

File: roles/k8s_best_practices_certsuite/tasks/tests.yml
Comment (ramperher) [reply roles/k8s_best_practices_certsuite/tasks/tests.yml:18]: Yes, it could be, this one I didn't modify it because it was defined in that way in cnf_cert role, now it's updated

**Review by tkrishtop — DISMISSED** [review]
LGTM

File: roles/cnf_cert/tasks/main.yml
Comment (reihl) [inline roles/cnf_cert/tasks/main.yml:5]: 'failed_when: false' will be much clean although fail will provide red text

Comment (ramperher) [reply roles/cnf_cert/tasks/main.yml:5]: done

File: ansible-collection-redhatci-ocp.spec
Comment (manurodriguez) [inline ansible-collection-redhatci-ocp.spec:54]: This date is fine, but if possible let's bump it up when ready to merge

**Review by manurodriguez — DISMISSED** [review]

File: ansible-collection-redhatci-ocp.spec
Comment (ramperher) [reply ansible-collection-redhatci-ocp.spec:54]: Bad thing is that the agent changes need to be pass Zuul check before merging, so I don't really know when I'm going to merge this :/ I'll keep an eye on this in the meantime

**Comment by ramperher** [comment]
@thekad , could you take a look to the change when you have some time to conclude the review? Thanks!

File: ansible-collection-redhatci-ocp.spec
Comment (thekad) [inline ansible-collection-redhatci-ocp.spec:57]: Add deprecation of cnf_cert role here

Comment (ramperher) [reply ansible-collection-redhatci-ocp.spec:57]: It is already in the message: _Version bump **due to deprecation of cnf_cert role**, moving in favour of k8s_best_practices_certsuite role_

File: ansible-collection-redhatci-ocp.spec
Comment (ramperher) [reply ansible-collection-redhatci-ocp.spec]: done

**Comment by ramperher** [comment]
All dependencies are ready for merging, waiting for the approval on this one to merge alltogether

**Review by thekad — APPROVED** [review]
:fleur_de_lis:

**Pull request merged**
